src/platform/unix/mod.rs

```diff
@@ -873,50 +924,37 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
     Ok((main_data_buffer, channels, shared_memory_regions))
 }
 
-#[cfg(target_os="android")]
-fn maybe_unlink(_: *const c_char) -> c_int {
-    // Calling `unlink` on a file stored on an sdcard immediately deletes it.
-    // FIXME: use a better temporary directory than the sdcard via the Java APIs
-    // and threading that value into Servo.
-    // https://code.google.com/p/android/issues/detail?id=19017
-    0
+unsafe fn create_filename() -> CString {
+    let mut tmp_string = TEMP_FILE_TEMPLATE.to_owned();
+    let addition: String = thread_rng().gen_ascii_chars().take(6).collect();
```

Comment:
I still have some work to do on this. In the meantime I removed the use of `thread_rng()` in this function and used a combination of the pid and timestamp. To be completely honest, I totally forgot to add the counter 😄 

> Seems more tricky for the POSIX SHM interface though

Yes, AFAIK both of the other platforms should be fine with the `lazy_static`, but it gets a lot trickier with any platforms using the `shm_open` implementation.